alpha/template/semver/semver.go

```diff
@@ -75,17 +75,42 @@ func readFile(reader io.Reader) (*semverTemplate, error) {
 		return nil, err
 	}
 
-	// default behavior is to generate only minor channels
-	sv := semverTemplate{
-		GenerateMajorChannels: false,
-		GenerateMinorChannels: true,
-	}
+	sv := semverTemplate{}
 	if err := yaml.UnmarshalStrict(data, &sv); err != nil {
 		return nil, err
 	}
+
 	if sv.Schema != schema {
 		return nil, fmt.Errorf("readFile: input file has unknown schema, should be %q", schema)
 	}
+
+	// if no generate option is selected, default to GenerateMinorChannels
+	if !sv.GenerateMajorChannels && !sv.GenerateMinorChannels {
+		sv.GenerateMinorChannels = true
+	}
+
+	// for default channel preference,
+	// if un-set, default to align to the selected generate option
+	// if set, error out if we mismatch the two
+	switch sv.DefaultChannelTypePreference {
+	case "":
```

Comment:
nit: I think it would be useful to have a constant here rather than "" (for readability =D)